pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3723,13 +3764,26 @@ private boolean writeLineageEntryWithTightLoop(String tableNameWithType, String
       if (!currentLineageEntry.equals(lineageEntryToMatch)) {
         String errorMsg = String.format(
             "Aborting the to update lineage entry since we find that the entry has been modified for table %s, "
-                + "entry id: %s", tableNameWithType, lineageEntryId);
+                + "entry id: %s", tableConfig, lineageEntryId);
         LOGGER.error(errorMsg);
         throw new RuntimeException(errorMsg);
       }
 
       // Update lineage entry
       segmentLineageToUpdate.updateLineageEntry(lineageEntryId, lineageEntryToUpdate);
+      switch (replaceSegmentType) {
+        case START:
```

Comment:
Should we throw exception in this case since it's not the valid path? Or, this is fine?

Reply:
We are not using `writeLineageEntryWithTightLoop()` for startReplacement api currently, but I think it's fine to have it there.